src/isomorphic/modern/class/__tests__/ReactES6Class-test.js

```diff
@@ -447,4 +449,62 @@ describe('ReactES6Class', () => {
     var node = ReactDOM.findDOMNode(instance);
     expect(node).toBe(container.firstChild);
   });
+
+  it('should warn when mutating the updater in the constructor', () => {
+    spyOn(console, 'error');
+
+    class Component extends React.Component {
+      constructor(props) {
+        super(props);
+        this.updater = {};
+      }
+
+      render() {
+        return null;
+      }
+    }
+
+    ReactDOM.render(<Component />, document.createElement('div'));
+
+    expect(console.error.calls.count()).toBe(1);
+    expect(console.error.calls.mostRecent().args).toMatchSnapshot();
+  });
+
+  it('should warn when mutating the updater somewhere else', () => {
+    spyOn(console, 'error');
+
+    class Component extends React.Component {
+      componentDidMount() {
+        this.updater = {};
+      }
+
+      render() {
+        return null;
+      }
+    }
+
+    ReactDOM.render(<Component />, document.createElement('div'));
+
+    expect(console.error.calls.count()).toBe(1);
+    expect(console.error.calls.mostRecent().args).toMatchSnapshot();
+  });
+
+  it('should not warn when mutating the updater with a valid update', () => {
+    spyOn(console, 'error');
+
+    class Component extends React.Component {
+      constructor(props) {
+        super(props);
+        this.updater = ReactNoopUpdateQueue;
```

Comment:
We prefer to test using only public APIs when possible. I don't think we need this test. If this were to fail, a bunch of other tests would fail too because we'd get an extra warning every time we created a class.